src/tokio/io.rs

```diff
@@ -204,10 +205,28 @@ where
             ctor_arg,
             underlying_io: tcp,
             options,
+            reconnection_stopped: false,
         })
     }
 
+    /// Stops any further reconnect attempts.
+    pub fn stop_reconnection(&mut self) {
```

Comment:
I would not want to force the user to have to do this by mutable self ref, perhaps an atomic bool would make more sense.